clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainStorageEngine.java

```diff
@@ -29,14 +29,15 @@
 import org.ehcache.clustered.common.internal.store.Util;
 import org.terracotta.offheapstore.paging.OffHeapStorageArea;
 import org.terracotta.offheapstore.paging.PageSource;
+import org.terracotta.offheapstore.storage.BinaryStorageEngine;
 import org.terracotta.offheapstore.storage.PointerSize;
 import org.terracotta.offheapstore.storage.StorageEngine;
 import org.terracotta.offheapstore.storage.portability.Portability;
+import org.terracotta.offheapstore.util.Factory;
 
 import static java.util.Collections.unmodifiableList;
 
-class OffHeapChainStorageEngine<K> implements StorageEngine<K, InternalChain> {
-
+public class OffHeapChainStorageEngine<K> implements ChainStorageEngine<K>, BinaryStorageEngine, StorageEngineExtensions {
```

Comment:
Something feels off about this class... I can't quite put my finger on it though.  There's something going on with the way we're mixing inlined composition, OO composition and then later OO inheritance that makes me feel uncomfortable.  I shall have a further think and see if I can explain this better.

Reply:
done changes as discussed